backend/DevUp/src/DevUp.Infrastructure.Postgres/Organization/Queries/Handlers/GetAllTeamsQueryHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Data;
+using AutoMapper;
+using Dapper;
+using DevUp.Application.Organization.Queries;
+using DevUp.Application.Organization.Queries.Results;
+using DevUp.Domain.Common.Extensions;
+using DevUp.Infrastructure.Postgres.Organization.Dtos;
+using DevUp.Infrastructure.Postgres.Setup;
+using MediatR;
+
+namespace DevUp.Infrastructure.Postgres.Organization.Queries.Handlers
+{
+    internal class GetAllTeamsQueryHandler : IRequestHandler<GetAllTeamsQuery, IEnumerable<TeamQueryResult>>
+    {
+        private readonly IDbConnection _connection;
+        private readonly IMapper _mapper;
+
+        public GetAllTeamsQueryHandler(IDbConnectionFactory connectionFactory, IMapper mapper)
+        {
+            _connection = connectionFactory.Create(DbConnectionName.Organization);
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<TeamQueryResult>> Handle(GetAllTeamsQuery request, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            if (request is null)
+                throw new ArgumentNullException(nameof(request));
+
+            var sql = @$"SELECT 
+                            id {nameof(TeamDto.Id)}, 
+                            name {nameof(TeamDto.Name)}
+                        FROM teams";
+
+            var dtos = await _connection.QueryAsync<TeamDto>(sql);
```

Comment:
Why are we directly calling to database here? Shouldn't that be part of `ITeamRepository.GetAllAsync`?

Reply:
I decided to not create `await .GetAllAsync()` in the team repository as realistically I don't see any reason for fetching all teams (at least at once, without any pagination which is not a domain concern). Notice that `.Handle()` for this query accepts query object - eventually this is where query filters will end up and it will just be easier (and probably more efficient, but that wasn't my goal) to bypass domain and apply them straight on SQL. This way I can make sure that filtering happens directly on the database.

One could argue that the same should happen for _all queries_ (for example `GetByIdAsync()`), but these are being used in domain services to ensure proper state (e.g. _you can't create team with id that is already taken_) and also probably won't be modified any time soon (and filters can and will expand).